src/GraphQL/Execution/ExecutionNode.cs

```diff
@@ -76,8 +76,19 @@ public object Source
         /// <summary>
         /// Initializes an instance of <see cref="ExecutionNode"/> with the specified values
         /// </summary>
+        /// <param name="parent">The parent node, or null if this is the root node</param>
+        /// <param name="graphType">The underlying graph type of this node; not a <see cref="NonNullGraphType"/>, and
+        /// for child nodes of an array execution node, not a <see cref="ListGraphType"/> instance. Typically
+        /// this should equal <see cref="FieldDefinition">fieldDefinition</see>.<see cref="FieldType.ResolvedType">ResolvedType</see>.<see cref="GraphQLExtensions.GetNamedType(IGraphType)">GetNamedType()</see></param>
+        /// <param name="field">The AST field of this node</param>
+        /// <param name="fieldDefinition">The graph's field type of this node</param>
+        /// <param name="indexInParentNode">For child array item nodes of a <see cref="ListGraphType"/>, the index of this array item within the field; otherwise, null</param>
         protected ExecutionNode(ExecutionNode parent, IGraphType graphType, Field field, FieldType fieldDefinition, int? indexInParentNode)
         {
+            if (graphType is NonNullGraphType)
+                throw new System.ArgumentOutOfRangeException(nameof(graphType), "For non-nullable graph types, graphType must represent the unwrapped graph type");
+            if (IndexInParentNode.HasValue && graphType is ListGraphType)
```

Comment:
What about list of lists ? 

[[Int]] : [[1], [1,2], [1,2,3]]

Reply:
I forgot that was possible.  I suggest we remove the checks, but keep the comments (after review to be sure they are still correct).

Reply:
OK